examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+    """Returns training and test images sampled randomly from the set of MNIST images.
+    Args:
+        train_size: Desired number of training images.
+        test_size: Desired number of test images.
+    Returns:
+        train_set: A list of length train_size containing images from the MNIST train dataset.
+        train_labels: Corresponding labels for the train images.
+        test_set: A list of length test_size containing images from the MNIST test dataset.
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
+
+    print("Fetched the data")
+    mnist_images = dataset["data"]
+    mnist_labels = dataset["target"].astype("int")
+
+    def _sample_data(images, labels, num_per_class):
+        t_set = []
+        t_labels = []
+        for i in range(10):
+            idxs = np.random.choice(np.argwhere(labels == i)[:, 0], num_per_class)
+            for idx in idxs:
+                img = images[idx].reshape(28, 28)
+                img_arr = jax.image.resize(
+                    image=img, shape=(112, 112), method="bicubic"
+                )
+                img = jnp.pad(
+                    img_arr,
+                    pad_width=tuple([(p, p) for p in (44, 44)]),
+                    mode="constant",
+                    constant_values=0,
+                )
+
+                t_set.append(img)
+                t_labels.append(i)
+        return t_set, t_labels
+
+    np.random.seed(seed)
+    full_train_set, full_train_labels = (
+        mnist_images[:mnist_train_size],
+        mnist_labels[:mnist_train_size],
+    )
+    full_test_set, full_test_labels = (
+        mnist_images[mnist_train_size:],
+        mnist_labels[mnist_train_size:],
+    )
+
+    train_set, train_labels = _sample_data(
+        full_train_set, full_train_labels, train_size // 10
+    )
+    test_set, test_labels = _sample_data(
+        full_test_set, full_test_labels, test_size // 10
+    )
+
+    return train_set, np.array(train_labels), test_set, np.array(test_labels)
+
+
+# %%
+train_set, train_labels, test_set, test_labels = fetch_mnist_dataset(
+    train_size, test_size
+)
+
+# %% [markdown]
+# # 2. Load the model
+
+# %%
+data = np.load("example_data/rcn.npz", allow_pickle=True, encoding="latin1")
+frcs, edges, suppression_masks, filters = (
+    data["frcs"],
+    data["edges"],
+    data["suppression_masks"],
+    data["filters"],
+)
+
+M = (2 * hps + 1) * (2 * vps + 1) + 1
+
+# %% [markdown]
+# # 3. Visualize loaded model
+
+# %%
+img = np.zeros((200, 200))
+
+frc, edge = frcs[4], edges[4]
+plt.figure(figsize=(10, 10))
+for e in edge:
+    i1, i2, w = e
+    f1, r1, c1 = frc[i1]
+    f2, r2, c2 = frc[i2]
+
+    img[r1, c1] = 255
+    img[r2, c2] = 255
+    plt.text((c1 + c2) // 2, (r1 + r2) // 2, str(w), color="blue")
+    plt.plot([c1, c2], [r1, r2], color="blue", linewidth=0.5)
+
+plt.imshow(img, cmap="gray")
```

Comment:
![Screenshot from 2021-11-20 16-11-50](https://user-images.githubusercontent.com/10180197/142744593-5d06f8a1-a268-4794-99ef-9aefa069eed7.png)
This visualization doesn't look good. Better to get rid of the boundary part, change coloring (I have lots of trouble reading the figure) by just using white background and use something other than blue for the perturb radiuses?

Reply:
Thanks! Looking much better. Can we further zoom in to the actual model and get rid of all the blank space around it (so `img[60:130, 60:160]` or something similar you see fit)

Reply:
Changed.